rust/gui-client/src-tauri/src/client/gui/system_tray/builder.rs

```diff
@@ -0,0 +1,222 @@
+//! An abstraction over Tauri's system tray menu structs, that implements `PartialEq` for unit testing
+
+use connlib_shared::messages::ResourceId;
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+/// A menu that can either be assigned to the system tray directly or used as a submenu in another menu.
+///
+/// Equivalent to `tauri::SystemTrayMenu`
+#[derive(Debug, Default, PartialEq, Serialize)]
+pub(crate) struct Menu {
+    pub(crate) entries: Vec<Entry>,
+}
+
+/// Something that can be shown in a menu, including text items, separators, and submenus
+///
+/// Equivalent to `tauri::SystemTrayMenuEntry`
+#[derive(Debug, PartialEq, Serialize)]
+pub(crate) enum Entry {
+    Item(Item),
+    Separator,
+    Submenu { title: String, inner: Menu },
+}
+
+/// Something that shows text and may be clickable
+///
+/// Equivalent to `tauri::CustomMenuItem`
+#[derive(Debug, PartialEq, Serialize)]
+pub(crate) struct Item {
+    /// An event to send to the app when the item is clicked.
+    ///
+    /// If `None`, then the item is disabled and greyed out.
+    pub(crate) event: Option<Event>,
+    /// The text displayed to the user
+    pub(crate) title: String,
+    /// A keyboard shortcut, e.g. Ctrl+Q for Quit.
+    ///
+    /// As far as we can tell these don't actually work.
```

Comment:
Why do we have them then? 🙃

Reply:
🤷‍♀️ Yeah I just tried on Debian+KDE and they don't work there... never seen them work on Windows...
Let's remove the keyboard accelerators so we don't give anyone false hope